kafka-on-docker/README.md

```diff
@@ -0,0 +1,145 @@
+Finally, you'll need an account on [docker hub](https://hub.docker.com/explore) so Docker Desktop can pull the images specified in the Docker Compose file.
+  
+## The Docker Compose file
+
+Before you get started, here's the Docker Compose file for the tutorial:
+
+```yaml
+services:
+  broker:
+    image: apache/kafka:latest
+    hostname: broker
+    container_name: broker
+    ports:
+      - 9092:9092
+    environment:
+      KAFKA_BROKER_ID: 1
+      KAFKA_LISTENER_SECURITY_PROTOCOL_MAP: PLAINTEXT:PLAINTEXT,PLAINTEXT_HOST:PLAINTEXT,CONTROLLER:PLAINTEXT
+      KAFKA_ADVERTISED_LISTENERS: PLAINTEXT://broker:29092,PLAINTEXT_HOST://localhost:9092
+      KAFKA_OFFSETS_TOPIC_REPLICATION_FACTOR: 1
+      KAFKA_GROUP_INITIAL_REBALANCE_DELAY_MS: 0
+      KAFKA_TRANSACTION_STATE_LOG_MIN_ISR: 1
+      KAFKA_TRANSACTION_STATE_LOG_REPLICATION_FACTOR: 1
+      KAFKA_PROCESS_ROLES: broker,controller
+      KAFKA_NODE_ID: 1
+      KAFKA_CONTROLLER_QUORUM_VOTERS: 1@broker:29093
+      KAFKA_LISTENERS: PLAINTEXT://broker:29092,CONTROLLER://broker:29093,PLAINTEXT_HOST://0.0.0.0:9092
+      KAFKA_INTER_BROKER_LISTENER_NAME: PLAINTEXT
+      KAFKA_CONTROLLER_LISTENER_NAMES: CONTROLLER
+      KAFKA_LOG_DIRS: /tmp/kraft-combined-logs
+      CLUSTER_ID: MkU3OEVBNTcwNTJENDM2Qk
+```
+Let's review some of the key parts of the YAML 
+
+```yaml
+ broker:
+    image: apache/kafka:latest
+    hostname: broker
+    container_name: broker
+    ports:
+      - 9092:9092
+```
+ - `image`  contains the organization, image name and the version to use.  In this case it will always pull the latest one available.
+ - `ports` specifies the ports you use to connect to Kafka.
+
+```yaml
+KAFKA_OFFSETS_TOPIC_REPLICATION_FACTOR: 1
+KAFKA_GROUP_INITIAL_REBALANCE_DELAY_MS: 0
+KAFKA_TRANSACTION_STATE_LOG_MIN_ISR: 1
+KAFKA_TRANSACTION_STATE_LOG_REPLICATION_FACTOR: 1
+```
+These settings affect topic replication and min-in-sync replicas and should only ever use these values when running a single Kafka broker on Docker. For a full explanation of these and other configurations,
+[consult the Kafka documentation](https://docs.confluent.io/platform/current/installation/configuration/broker-configs.html#cp-config-brokers).
+
+```yaml
+KAFKA_PROCESS_ROLES: broker,controller
+```
+
+Since the introduction of [KRaft]() Kafka no longer uses [Apache ZooKeeper®](https://zookeeper.apache.org/) for managing cluster metadata, using Kafka itself instead.  One advantage of the new KRaft mode is that you can have a single Kafka broker to handle both metadata and handling client requests in small, local development environment.  The `docker-compose.yml` file for this tutorial uses this approach leading to faster startup times and simpler configuration.  Note that in a production setting you'll have distinct Kafka brokers for handling requests and operating as a cluster controller.
```

Comment:
```suggestion
Since the introduction of [KRaft](https://docs.confluent.io/platform/current/kafka-metadata/kraft.html) Kafka no longer requires [Apache ZooKeeper®](https://zookeeper.apache.org/) for managing cluster metadata, using Kafka itself instead.  One advantage of the new KRaft mode is that you can have a single Kafka broker to handle both metadata and client requests in small, local development environment.  The `docker-compose.yml` file for this tutorial uses this approach, leading to faster startup times and simpler configuration.  Note that, in a production setting, you'll have distinct Kafka brokers for handling requests and operating as a cluster controller.
```